tracer/tracer.go

```diff
@@ -201,7 +195,7 @@ func NewTracer(ctx context.Context, cfg *Config) (*Tracer, error) {
 	hasBatchOperations := ebpfHandler.SupportsGenericBatchOperations()
 
 	processManager, err := pm.New(ctx, cfg.IncludeTracers, cfg.Intervals.MonitorInterval(),
-		ebpfHandler, nil, cfg.Reporter, elfunwindinfo.NewStackDeltaProvider(),
+		ebpfHandler, nil, nil, elfunwindinfo.NewStackDeltaProvider(),
```

Comment:
Instead of passing `nil` as `reporter.ExecutableReporter`, maybe pass a dummy implementation that satisfies just the interface.

The worry is, that before every use of `reporter.ExecutableReporter` we would need a `nil` check - which clumbers the code and will be likely forgetten at some point. Until execution path reaches this code and panics.

Reply:
The executable metadata for reporter is now directly linked to the `libpf.Frame` and can be accessed from it it. The `exeReporter` is now only used as a hook for not-in-tree repoters that use it to potentially upload executables (or other files related to it such as `.debug` files) to the backend for symbolization purposes.

Reply:
Does edac136fd598cccbad36a6c81e0f4e8d23335539 address these sufficiently?

Reply:
Removing `ExecutableReporter` from `Config` makes it impossible for Tracer's clients to pass it a reporter.

Reply:
@fabled Currently, this breaks our ability to pass a custom `ExecutableReporter` to the Tracer, unless we're missing something.

Reply:
Seems this was unfortunate oversight in how the reporter is passed in the other implementations. Seems PR #747 adds this back. But to get this fixed sooner a PR just adding this back is welcome. I can also do it in a little bit if preferred.

Reply:
Is #748 sufficient?